src/sync.js

```diff
@@ -116,3 +138,15 @@ export class SyncWalker extends WalkerBase {
 		return node;
 	}
 }
+
+/**
+ * Ducktype a node.
+ *
+ * @param {unknown} value
+ * @returns {value is Node}
+ */
+function isNode(value) {
+	return (
+		value !== null && typeof value === 'object' && 'type' in value && typeof value.type === 'string'
+	);
+}
```

Comment:
This function is defined twice. (It’s also in `async.js`.) I suggest to deduplicate this logic.

Reply:
a lot of the code is duplicated tho